unity-renderer/Assets/Desktop/Scripts/MainScripts/DCL/Controllers/HUD/Preloading/PreloadingController.cs

```diff
@@ -13,34 +15,40 @@ public class PreloadingController : IDisposable
         private readonly DataStoreRef<DataStore_LoadingScreen> loadingScreenRef;
 
         private BaseVariable<bool> isSignUpFlow => DataStore.i.common.isSignUpFlow;
-        private bool isDisposed = false;
+        private bool isDisposed;
 
-        public PreloadingController()
+        private CancellationTokenSource cancellationTokenSource;
+        public PreloadingController(IAddressableResourceProvider addressableResourceProvider)
```

Comment:
1. It looks like it should be a service and not a new instance
2. So it will be resolved via `Service<T>`
3. `GetView(cancellationTokenSource.Token)` will be moved to `Initialize()` then

Reply:
It was a problem of order of initialization, so the service wasn't created yet. Now I moved it after services initialization and getting it inside the constructor. However, I still do prefer to use constructor injection for required dependencies and property injection only for optional dependencies (as it is well-known best practice of DI).